core/src/main/java/org/apache/calcite/util/Sources.java

```diff
@@ -189,14 +196,25 @@ private static File urlToFile(URL url) {
       return Paths.get(uri).toFile();
     }
 
+    private static URL fileToUrl(File file) {
+      URI uri;
+      try {
+        uri = file.toURI();
+      } catch (SecurityException e) {
+        throw new IllegalArgumentException("No access to the underlying file " + file.getPath());
```

Comment:
Why do you omit `SecurityException e` in the caused by chain?

Reply:
Nice catch. Thank you. 
I have revised the code to wrap the SecurityException in the IllegalArgumentException.